### PR TITLE
HISAT2 aligner

### PR DESCRIPTION
**Pull request by ewels**
Most of the basic code for running with HISAT2 and building its required references is now written. It seems to compile and start on some of the first processes. There's almost certainly stuff broken, but I figured if I create a pull-request then we can share the testing as it's slow.

### CODE REVIEW

**Comment by Hammarn** [comment]
Looks real nice, good job. Does it run at all?

**Comment by ewels** [comment]
It runs a little :) Testing more and bugfixing now, will get back to you when I think it's ready for a final test / merge.

**Comment by ewels** [comment]
Ok @Hammarn - ready to merge when you're happy (would be great if you could run a test again, haven't run it to completion with more than one file for a while).

**Comment by Hammarn** [comment]
Ok I've tested and the Pipeline ran to completion on Milou with human data, so I'm happy to merge this! 👍

File: main.nf [hunk 1]
```diff
@@ -314,11 +402,103 @@ def check_log(logs) {
...
+/*
+ * STEP 3 - align with HISAT2
+ */
+if(params.aligner == 'hisat2'){
+    process hisat2 {
+        tag "$reads"
+        publishDir "${params.outdir}/HISAT2", mode: 'copy'
+
+        cpus { params.star_cpus ?: 10 }
+        memory { (params.star_memory ?: 80.GB) * task.attempt }
```
Comment (ewels) [inline main.nf]: I've actually removed `params.star_memory` in the next pull request that I'm about to open 😉 

But yes, we need to do proper profiling soon...

**Pull request merged**